File: spryntax/website_static_build_tools/alg_creator/algs-py/radixsort.py
def counting_sort(arr, exp):
    n = len(arr)
    output = [0] * n
    count = [0] * 10

    # Store the count of occurrences in count[]
    for i in range(n):
        index = arr[i] // exp
        count[index % 10] += 1

    # Change count[i] so that count[i] contains actual position of this digit
    for i in range(1, 10):
        count[i] += count[i - 1]

    # Build the output array
    i = n - 1
    while i >= 0:
        index = arr[i] // exp
        output[count[index % 10] - 1] = arr[i]
        count[index % 10] -= 1
        i -= 1

    # Copy the sorted numbers into original array
    for i in range(n):
        arr[i] = output[i]

# Main function to implement Radix Sort
def radix_sort(arr):
    # Find the maximum number to determine number of digits
    max_num = max(arr)
    exp = 1  # Start with the least significant digit

    # Perform counting sort for every digit
    while max_num // exp > 0:
        counting_sort(arr, exp)
        exp *= 10

# Sample input
arr = [170, 45, 75, 90, 802, 24, 2, 66]
radix_sort(arr)
print("Sorted array:", arr)

File: spryntax/website_static_build_tools/alg_creator/algs-c/radixsort.c
#include <stdio.h>

// Function to perform counting sort for radix sort
void counting_sort(int arr[], int n, int exp) {
	int output[n], count[10] = {0};

	// Store the count of occurrences in count[]
	for (int i = 0; i < n; i++) {
		int index = arr[i] / exp;
		count[index % 10]++;
	}

	// Change count[i] so that count[i] contains actual position of this digit
	for (int i = 1; i < 10; i++) {
		count[i] += count[i - 1];
	}

	// Build the output array
	for (int i = n - 1; i >= 0; i--) {
		int index = arr[i] / exp;
		output[count[index % 10] - 1] = arr[i];
		count[index % 10]--;
	}

	// Copy the sorted numbers into the original array
	for (int i = 0; i < n; i++) {
		arr[i] = output[i];
	}
}

// Main function to implement Radix Sort
void radix_sort(int arr[], int n) {
	// Find the maximum number to determine the number of digits
	int max_num = arr[0];
	for (int i = 1; i < n; i++) {
		if (arr[i] > max_num) {
			max_num = arr[i];
		}
	}

	// Perform counting sort for every digit
	for (int exp = 1; max_num / exp > 0; exp *= 10) {
		counting_sort(arr, n, exp);
	}
}

int main() {
	// Sample input
	int arr[] = {170, 45, 75, 90, 802, 24, 2, 66};
	int n = sizeof(arr) / sizeof(arr[0]);

	radix_sort(arr, n);
	printf("Sorted array: ");
	for (int i = 0; i < n; i++) {
		printf("%d ", arr[i]);
	}
	printf("\n");

	return 0;
}
